### PR TITLE
*: show memory usage in "show processlist"

### PR DESCRIPTION
**Pull request by zz-jason**
after this PR:
```sql
TiDB(localhost) > show processlist;
+------+------+-----------+------+---------+------+-------+--------------------------------+----------+
| Id   | User | Host      | db   | Command | Time | State | Info                           | Mem      |
+------+------+-----------+------+---------+------+-------+--------------------------------+----------+
|    3 | root | 127.0.0.1 | test | Query   |    0 | 2     | show processlist               |        0 |
|    1 | root | 127.0.0.1 | test | Query   |  162 | 2     | select * from t t1, t t2, t t3 | 19906560 |
+------+------+-----------+------+---------+------+-------+--------------------------------+----------+
2 rows in set (0.01 sec)
```

### CODE REVIEW

**Comment by zz-jason** [comment]
@winoros @XuHuaiyu @shenli PTAL

**Comment by winkyao** [comment]
CI failed

File: session/session.go [hunk 1]
```diff
@@ -1380,6 +1380,7 @@ func (s *session) ShowProcess() util.ProcessInfo {
 	tmp := s.processInfo.Load()
 	if tmp != nil {
 		pi = tmp.(util.ProcessInfo)
+		pi.Mem = s.GetSessionVars().StmtCtx.MemTracker.BytesConsumed()
```
Comment (lamxTyler) [inline session/session.go]: Could we do it in `SetProcessInfo`?

Comment (zz-jason) [reply session/session.go]: No we can not. Memory usage is dynamically changed value, we should calculate the memory usage when "show processlist" is called not when a query is being executed.

**Pull request merged**